### PR TITLE
BaseTools: Remove Pip BaseTools

### PR DESCRIPTION
**Pull request by os-d**
# Description

BaseTools was moved out to a separate repo and consumed as a pip module by edk2 CI. This process has not led to the desired goals of doing so, so this patch removes the pip based BaseTools from edk2 CI.

The original goal of moving BaseTools to a pip module was primarily to speed up the development process, as the old edk2 mailing list was slow. However, with edk2 moving to PRs, it now actually slows the BaseTools development process to have to do a PR in another repo, publish the module, and then make a PR in edk2 to consume the new BaseTools. It also holds up using the features in a new BaseTools in other PRs.

There were other goals of moving, such as allowing projects to use the BaseTools outside of edk2. This can still be accomplished outside of this PR, this PR simply stops edk2 CI from using the pip module.

This is marked as a breaking change as consumers may be moving forward in BaseTools or may need to build the BaseTools before doing another build.

For more discussion on this, see this PR's comments that led to this PR: https://github.com/tianocore/edk2/pull/5957#issuecomment-2327635989.

- [x] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

## How This Was Tested

Tested with local CI and pipeline CI.

## Integration Instructions

If local BaseTools have not been built, consumers will need to build them before doing another build by doing:

```bash
python BaseTools/Edk2ToolsBuild.py -t <TOOL_CHAIN_TAG>
```

### CODE REVIEW

File: BaseTools/BinPipWrappers/PosixLike/BPDG
Comment (bcran) [inline BaseTools/BinPipWrappers/PosixLike/BPDG:1]: Why are all these files being deleted?

Comment (os-d) [reply BaseTools/BinPipWrappers/PosixLike/BPDG:1]: @bcran , these files were all added to support the pip basetools: https://github.com/tianocore/edk2/commit/1534b6228b0932d745bf6ec9fd4cb010800b92f2. So as part of removing the pip basetools, this cleans up these now unused files.

Comment (os-d) [reply BaseTools/BinPipWrappers/PosixLike/BPDG:1]: There were also two BaseTools build scripts edited in that commit that I missed, I just pushed an update to convert those back to in-tree only.

Comment (bcran) [reply BaseTools/BinPipWrappers/PosixLike/BPDG:1]: Ah, yes - `BinPipWrappers` not `BinWrappers`!

**Review by bcran — APPROVED** [review]

**Review by Javagedes — APPROVED** [review]

**Comment by os-d** [comment]
@mdkinney , we have .pytool and BaseTools maintainer approval, did you have any comment? If not, @bcran , can we add the push label?

**Comment by bcran** [comment]
@mdkinney Could you or somebody else with admin rights archive the edk2-basetools repository if it's no longer going to be used please?

**Comment by lgao4** [comment]
> @mdkinney Could you or somebody else with admin rights archive the edk2-basetools repository if it's no longer going to be used please?

Can we merge this change first?

File: .pytool/CISettings.py [hunk 1]
```diff
@@ -46,8 +45,6 @@ def __init__(self):
 
     def AddCommandLineOptions(self, parserObj):
         group = parserObj.add_mutually_exclusive_group()
```
Comment (spbrogan) [inline .pytool/CISettings.py:47]: is the group still needed.  I don't see any other parameter using it and i don't know what happens if you create a mutually exclusive group but then never add any arguments.

**Review by spbrogan — COMMENTED** [review]
I

File: .pytool/CISettings.py (hunk 1 above)
Comment (os-d) [reply .pytool/CISettings.py:47]: I'll remove it.

**Review by spbrogan — APPROVED** [review]

**Comment by os-d** [comment]
@mdkinney, do you want to review before this gets merged? Otherwise it has the appropriate approvals

**Pull request merged**